src/components/member-page/donation/table/row.js

```diff
@@ -0,0 +1,208 @@
+  }
+  ${mq.mobileOnly`
+    display: none;
+  `}
+`
+
+const P1Gray800 = styled(P1)`
+  color: ${colorGrayscale.gray800};
+`
+
+const P1Gray600 = styled(P1)`
+  color: ${colorGrayscale.gray600};
+`
+
+const MobileTableContent = styled.div`
+  width: 100%;
+  display: flex;
+  flex-direction: column;
+  align-items: flex-start;
+  gap: 8px;
+  align-self: stretch;
+  .row {
+    width: 100%;
+    display: flex;
+    gap: 8px;
+    .amount {
+      display: flex;
+      flex-grow: 1;
+      justify-content: end;
+    }
+  }
+  .donation-number {
+    width: 100%;
+    display: block;
+    word-break: break-all;
+  }
+  ${mq.tabletAndAbove`
+    display: none;
+  `}
+`
+
+const ArrowIcon = styled(Arrow)`
+  transform: rotate(${props => props.rotate});
+  transition: all 300ms;
+`
+
+const Divider = styled(divider)`
+  width: 100%;
+  grid-column: 1 / 7;
+  margin-top: 16px;
+  margin-bottom: 16px;
+`
+
+export const formattedDate = date =>
+  `${date.getUTCFullYear()}/${date.getUTCMonth() + 1}/${date.getUTCDate()}`
+
+export const TableRow = memo(({ record }) => {
+  const {
+    type,
+    created_at: createdAt,
+    amount,
+    status,
+    order_number: orderNumber,
+  } = record
+  const donationDate = formattedDate(new Date(createdAt))
+  const amountSuffix = type === DonationType.PRIME ? '元' : '元/月'
+  const [isOpen, setIsOpen] = useState(false)
+  const [isLoading, setIsLoading] = useState(true)
+  const [periodicHistory, setPeriodicHistory] = useState([])
+  const { releaseBranch } = useContext(CoreContext)
+  const [state, dispatch] = useStore()
+
+  const getPeriodicDonationHistory = () => {
+    setIsLoading(true)
+    const detail = _.get(state, [
+      reduxStateFields.donationHistory,
+      'periodicDonationHistory',
+      'records',
+      orderNumber,
+    ])
+    if (detail) {
+      setPeriodicHistory(detail.records)
+    } else {
+      const jwt = _.get(state, [reduxStateFields.auth, 'accessToken'])
+      dispatch(getUserPeriodicDonationHistory(jwt, orderNumber, 0, 18)).then(
+        res => {
+          setPeriodicHistory(_.get(res, 'payload.data.records'))
+        }
+      )
+    }
+    setTimeout(() => setIsLoading(false), 1000)
+  }
+
+  const handleRowClick = () => {
+    if (type !== DonationType.PERIODIC) return
+    if (!isOpen) {
+      getPeriodicDonationHistory()
+    }
+    setIsOpen(!isOpen)
+  }
+  return (
+    <React.Fragment>
+      <TableContent onClick={handleRowClick}>
+        <P1Gray800 className="donation-date" text={donationDate} />
+        <P1Gray800 className="type" text={DonationTypeLabel[type]} />
+        <P1Gray800 className="donation-number" text={orderNumber} />
+        <P1Gray800
+          className="amount"
+          text={`${amount.toLocaleString('en-US')}${amountSuffix}`}
+        />
+        <div className="status">
+          <StatusBadge status={status} type={type} />
+          {type === DonationType.PERIODIC && (
+            <ArrowIcon
+              direction="down"
```

Comment:
same as [this comment ](https://github.com/twreporter/twreporter-react/pull/2823/files#r1568312757)

Reply:
fixed